code/modules/mob/living/simple_animal/abnormality/he/singing_machine.dm

```diff
@@ -32,34 +39,52 @@ Finally, an abnormality that DOESN'T have to do any fancy movement shit. It's a
 
 	pixel_x = -8
 	base_pixel_x = -8
+	pixel_y = -6
+	base_pixel_y = -6
 	buckled_mobs = list()
 	buckle_lying = TRUE
 	max_boxes = 16
-	var/cleanliness = "clean"
-	var/statChecked = 0
+	var/statCheckPenalty = 0
 	var/bonusRed = 0
 	var/grindRed = 4
 	var/minceRed = 8
+	var/clampRed = 80
+	var/playWhite = 50
 	var/playTiming = 5 SECONDS
 	var/playLength = 60 SECONDS
-	var/playStatus = 0
-	var/playRange = 20
+	var/playStatus = SILENT
+	var/playRange = 15
 	var/noiseFactor = 2
+	var/list/musicalAddicts = list()
+
+	//Sound Vars
 	var/datum/looping_sound/singing_grinding/grindNoise
 	var/datum/looping_sound/singing_music/musicNoise
-	var/list/musicalAddicts = list()
+
+	//Visual Vars
+	var/obj/effect/singing_machine_hinge/machine_hinge
+	var/obj/particle_emitter/singing_note/particle_note
+	var/obj/particle_emitter/singing_note_broken/particle_note_broken
+	var/obj/particle_emitter/singing_grind/particle_grind
+	var/obj/particle_emitter/singing_blood/particle_blood
+
+/mob/living/simple_animal/hostile/abnormality/singing_machine/Initialize()
+	..()
+	machine_hinge = new/obj/effect/singing_machine_hinge(src)
+	vis_contents += machine_hinge
+
+/mob/living/simple_animal/hostile/abnormality/singing_machine/Destroy()
+	QDEL_NULL(machine_hinge)
+	..()
 
 /mob/living/simple_animal/hostile/abnormality/singing_machine/Life()
-	if(playStatus > 0) // If playstatus isn't 0, deal some damage in range.
-		for(var/mob/living/carbon/human/H in livinginrange(playRange, src))
-			if(faction_check_mob(H))
-				continue
-			H.apply_damage(rand(playStatus * noiseFactor, playStatus * noiseFactor * 2), WHITE_DAMAGE, null, H.run_armor_check(null, WHITE_DAMAGE), spread_damage = TRUE)
-			if(H in musicalAddicts)
-				H.apply_damage(rand(playStatus * noiseFactor, playStatus * noiseFactor * 2), WHITE_DAMAGE, null, H.run_armor_check(null, WHITE_DAMAGE), spread_damage = TRUE)
-				to_chat(H, span_warning("You can hear it again... it needs more..."))
-			else
-				to_chat(H, span_warning("That terrible grinding noise..."))
+	if(playStatus == SILENT) // If the machine isn't silent, deal damage to everyone in range.
+		return ..()
+	noiseEffect()
+
+	if(!playStatus == PLAYING) // If the machine is playing music, also deal damage to all addicts and try to make them go insane.
```

Comment:
```suggestion
	if(playStatus != PLAYING) // If the machine is playing music, also deal damage to all addicts and try to make them go insane.
```
`!variable_name == value` is NOT equivalent to `variable_name != value`, it's equivalent to `!2 == 5` which makes no sense even if it may work.